geocruncher/main.py

```diff
@@ -64,6 +64,20 @@ def run_geocruncher(args):
         # TODO do something useful with output files
         print(generated_mesh_paths)
 
+    if args[1] == 'slice':
+        nPoints=500
+        data = json.loads(args[2])
+        xCoord=np.array([data["start"]["x"], data["end"]["x"]])
+        yCoord=np.array([data["start"]["y"], data["end"]["y"]])
+        zCoord=np.array([data["minElevation"], data["maxElevation"]])  

```

Comment:
Can we actually use the same encoding here for the slice rectangle as we use it for the sections? So having lowerLeft and upperRight with x/y/z? Or would that not work here?

Reply:
@phile314-fh That would work but then we would have to "sort" the vertex before.

Here I call them start because they really are the start of the line and this is because of this that this is needed:

`        xCoordinds = (-xCoord).argsort() #reverse order sort
        xCoord = xCoord[xCoordinds[::-1]]
        yCoord = yCoord[xCoordinds[::-1]]
`

We then go from start end to lowerLeft and upperRight. But this could also easily be done within scala if you want.

Reply:
I don't understand why we would need to sort. ``lowerLeft`` is the lower left coordinate of the slice rectangle:
lowerLeft ~=  (``start["x"], start["y"], minElevation)
upperRight ~=  (``end["x"], end["y"], maxElevation)

I think this shouldn't really affect any logic, just make the interface more consistent, no?

Reply:
You did not understood my talk: 

We already sort with this small code I pasted. I just stated why it is called start: because this is the starting point of the slice: namely the first point we have picked.

Obviously the first point we have picked might not be the upperLeft one. So what I say is:

We have a starting point and an end point. We can sort the to be left and right at any point of the code (even in the frontEnd). However I choosed to keep the as is and sort them in python. Would you like me to change that?

Also: in addition upper and lower make little sense here as we have only a x coordinates.

Reply:
Well, we do have z-coordinates, they just happen to match the project min/max elevation. And the name lowerLeft would just be a synonym for start, as it is the lowerLeft corner of the slice rectangle. So sorting would not be necessary. I mean we have exactly the same for sections, and we don't sort there either.

Anyhow, as this is only temporary, it is probably not worth the effort to change this.

Reply:
I guess the lower-left confusion is due to different reference systems. The lower left itself is relative to the slice rectangle, but specified in project coordinates.